netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```
@@ -655,7 +684,11 @@ public void setSslEngine(SSLEngine sslEngine) {
     }
 
     private void verifyAuthorityAllowedForPeerCert(String authority)
-        throws SSLPeerUnverifiedException, CertificateException {
+            throws SSLPeerUnverifiedException, CertificateException, InvocationTargetException,
+            IllegalAccessException {
+      if (checkServerTrustedMethod == null) {
+        throw new IllegalStateException("Method checkServerTrusted not found in javax.net.ssl.X509ExtendedTrustManager");
```

This exception is not handled by the multi-catch, right? Shouldn't it be?

---

Added now.

---

On 2nd thoughts there is no real value in having a warning log for an impossible exception and to check for the reflected method being null even though X509ExtendedTrustManager was found in class path (and both X509ExtendedTrustManager class and the checkServerTrusted method are available since API level 24), so I have removed these code.